### PR TITLE
Adding logging to Darwin

### PR DESCRIPTION
**Pull request by woody-apple**
Adding logging

### CODE REVIEW

**Review by anush-apple — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTRDevice.mm [hunk 1]
```diff
@@ -844,11 +844,13 @@ - (void)_addDelegate:(id<MTRDeviceDelegate>)delegate queue:(dispatch_queue_t)que
 #endif
 
     if (shouldSetUpSubscription) {
+        MTR_LOG("%@ - starting subscription setup", self);
         // Record the time of first addDelegate call that triggers initial subscribe, and do not reset this value on subsequent addDelegate calls
         if (!_initialSubscribeStart) {
             _initialSubscribeStart = [NSDate now];
         }
         if ([self _deviceUsesThread]) {
+            MTR_LOG(" => %@ - device is a thread device, scheduling in pool");
```
Comment (woody-apple) [inline src/darwin/Framework/CHIP/MTRDevice.mm:853]:
```suggestion
            MTR_LOG(" => %@ - device is a thread device, scheduling in pool", self);
```

File: src/darwin/Framework/CHIP/MTRDevice.mm [hunk 2]
```diff
@@ -1310,6 +1312,7 @@ - (void)_scheduleSubscriptionPoolWork:(dispatch_block_t)workBlock inNanoseconds:
         // In the case where a resubscription triggering event happened and already established, running the work block should result in a no-op
         MTRAsyncWorkItem * workItem = [[MTRAsyncWorkItem alloc] initWithQueue:self.queue];
         [workItem setReadyHandler:^(id _Nonnull context, NSInteger retryCount, MTRAsyncWorkCompletionBlock _Nonnull completion) {
+            MTR_LOG("%@ - work item is ready to attempt pooled subscription");
```
Comment (woody-apple) [inline src/darwin/Framework/CHIP/MTRDevice.mm:1315]:
```suggestion
            MTR_LOG("%@ - work item is ready to attempt pooled subscription", self);
```

File: src/darwin/Framework/CHIP/MTRDevice.mm [hunk 3]
```diff
@@ -177,8 +177,8 @@ - (void)_deviceInternalStateChanged:(MTRDevice *)device;
         UnsolicitedMessageFromPublisherHandler unsolicitedMessageFromPublisherHandler, ReportBeginHandler reportBeginHandler,
         ReportEndHandler reportEndHandler)
         : MTRBaseSubscriptionCallback(attributeReportCallback, eventReportCallback, errorCallback, resubscriptionCallback,
-            subscriptionEstablishedHandler, onDoneHandler, unsolicitedMessageFromPublisherHandler, reportBeginHandler,
-            reportEndHandler)
+              subscriptionEstablishedHandler, onDoneHandler, unsolicitedMessageFromPublisherHandler, reportBeginHandler,
+              reportEndHandler)
```
Comment (woody-apple) [inline src/darwin/Framework/CHIP/MTRDevice.mm:181]:
```suggestion
            subscriptionEstablishedHandler, onDoneHandler, unsolicitedMessageFromPublisherHandler, reportBeginHandler,
            reportEndHandler)
```

File: src/darwin/Framework/CHIP/MTRError_Internal.h [hunk 4]
```diff
@@ -26,6 +26,10 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+#ifndef YES_NO
+#define YES_NO(x) ((x) ? @"YES" : @"NO")
```
Comment (andy31415) [inline src/darwin/Framework/CHIP/MTRError_Internal.h:30]: This seems added but never used.

readability-wise, is this typical style? reading code of `YES_NO(foo)` is not immediately obvious for being a NSSTRING ... maybe `YES_NO_STR` or something like that would be more obvious?

**Review by andy31415 — APPROVED** [review]

**Review by kiel-apple — APPROVED** [review]

**Pull request merged**